core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala

```diff
@@ -37,6 +45,9 @@ private[spark] class ShuffleMapStage(
 
   override def toString: String = "ShuffleMapStage " + id
 
+  /** Map-stage jobs that were submitted to execute this stage independently (if any) */
```

Comment:
can you expand that this only for jobs that that were submitted *just* for this stage?  Ie., it does *not* include jobs that indirectly depend on this stage. 
Also, this is kinda obvious from the type, but do you mind expanding the comment to say that this only *running* jobs that were submitted for this job?  So that if  job completes, its removed from this list.